### PR TITLE
Prevent creating a project when no system is installed, closes #147

### LINKED ISSUE #147: Prevent creating a project if no system is installed.
Someone in French Skype group had this issue:

> " The server crashed while loading project "artemyx".
> Error: The system  is not installed."

It looks like we might be allowing people to create a project with no system selected at all, probably  when they declined installing Superpowers Game. We should prevent creating a project if no system is installed.

### PR DESCRIPTION
**Pull request by Pangoraw**

### CODE REVIEW

**Comment by elisee** [comment]
Hey!

Thanks for the PR. Will review it now!

File: client/src/hub/index.ts [hunk 1]
```diff
@@ -195,12 +195,16 @@ function onProjectActivate() {
...
 let autoOpenProject = true;
 function onNewProjectClick() {
-  new CreateOrEditProjectDialog(data.systemsById, { autoOpen: autoOpenProject }, (result) => {
-    if (result == null) return;
-    autoOpenProject = result.open;
-
-    socket.emit("add:projects", result.project, onProjectAddedAck);
-  });
+  if (Object.keys(data.systemsById).length === 0) {
+    new SupClient.Dialogs.InfoDialog(SupClient.i18n.t("hub:newProject.noSystemError.message"), { header: SupClient.i18n.t("hub:newProject.noSystemError.header")});
```
Comment (elisee) [inline client/src/hub/index.ts]: Missing a space before the closing `}` at the end

File: public/locales/en/hub.json [hunk 2]
```diff
@@ -2,6 +2,9 @@
   "title": "Hub",
   "serverAddress": "${hostname} on port ${port}",
   "newProject": {
+    "noSystemError": {
+      "header": "Can't create a project",
+      "message": "This server doesn't have any systems installed yet. Please install one from the Server Settings tab first."},
```
Comment (elisee) [inline public/locales/en/hub.json]: The closing bracket `}` should be on its own line here.

File: public/locales/fr/hub.json [hunk 3]
```diff
@@ -1,6 +1,10 @@
 {
   "serverAddress": "${hostname} sur le port ${port}",
   "newProject": {
+    "noSystemError": {
+      "header": "Impossible de créer un projet",
+      "message": "Ce serveur n'a aucun système installé. Veuillez premièrement en installer un depuis l'onglet de paramètrage du serveur."
```
Comment (elisee) [inline public/locales/fr/hub.json]: I'd replace "premièrement" with "d'abord"

File: public/locales/fr/hub.json [hunk 4]
```diff
@@ -17,4 +21,4 @@
   },
   "openProject": "Ouvrir le projet",
   "language": "Langue"
-}
+}
```
Comment (elisee) [inline public/locales/fr/hub.json]: There should be a new line at the end of the file.

**Comment by elisee** [comment]
Looks good overall, I left a few comments for minor things to fix.

**Comment by Pangoraw** [comment]
Thanks for reviewing ! I got it updated !

**Comment by elisee** [comment]
Merged, thanks!

**Pull request merged**